test/transforms.jl

```diff
@@ -673,6 +673,89 @@
     @test ttypes == Tables.schema(tₒ).types    
   end
 
+  @testset "Replace" begin
+    a = [3, 2, 1, 4, 5, 3]
+    b = [2, 4, 4, 5, 8, 5]
+    c = [1, 1, 6, 2, 4, 1]
+    d = [4, 3, 7, 5, 4, 1]
+    e = [5, 5, 2, 6, 5, 2]
+    f = [4, 4, 3, 4, 5, 2]
+    t = Table(; a, b, c, d, e, f)
+
+    # replace with a value of the same type
+    T = Replace(1 => -1, 5 => -5)
+    n, c = apply(T, t)
+    @test n.a == [3, 2, -1, 4, -5, 3]
+    @test n.b == [2, 4, 4, -5, 8, -5]
+    @test n.c == [-1, -1, 6, 2, 4, -1]
+    @test n.d == [4, 3, 7, -5, 4, -1]
+    @test n.e == [-5, -5, 2, 6, -5, 2]
+    @test n.f == [4, 4, 3, 4, -5, 2]
+    @test isrevertible(T) == true
+    tₒ = revert(T, n, c)
+    @test t == tₒ
+
+    # table schema after apply and revert
+    T = Replace(1 => -1, 5 => -5)
+    n, c = apply(T, t)
+    types = Tables.schema(t).types
+    @test types == Tables.schema(n).types
+    tₒ = revert(T, n, c)
+    @test types == Tables.schema(tₒ).types
+
+    # replace with a value of another type
+    T = Replace(1 => 1.5, 5 => 5.5, 4 => true)
+    n, c = apply(T, t)
+    @test n.a == Real[3, 2, 1.5, true, 5.5, 3]
+    @test n.b == Real[2, true, true, 5.5, 8, 5.5]
+    @test n.c == Real[1.5, 1.5, 6, 2, true, 1.5]
+    @test n.d == Real[true, 3, 7, 5.5, true, 1.5]
+    @test n.e == Real[5.5, 5.5, 2, 6, 5.5, 2]
+    @test n.f == Real[true, true, 3, true, 5.5, 2]
+    tₒ = revert(T, n, c)
+    @test t == tₒ
+
+    # table schema after apply and revert
+    T = Replace(1 => 1.5, 5 => 5.5, 4 => true)
+    n, c = apply(T, t)
+    tₒ = revert(T, n, c)
+    ttypes = Tables.schema(t).types
+    ntypes = Tables.schema(n).types
+    @test ntypes[1] == typejoin(typeof.(n.a)...)
```

Comment:
Isn't this the eltype of n.a? Can we just use eltype?

Reply:
`n.a` has values with different types.
In this case:
```julia
a = Real[3, 2, 1.5, true, 5.5, 3]
typejoin(typeof.(a)...)
typejoin(Int, Int, Float64, Bool, Float64, Int)
Real
```
It's just to make sure the common type between column values ​​is correct.

Reply:
Yes, but eltype(n.a) returns Real as well. What am I missing?

Reply:
`eltype(n.a)` could be `Number` or `Any`:
```julia
a = Any[3, 2, 1.5, true, 5.5, 3]
typejoin(typeof.(a)...)
typejoin(Int, Int, Float64, Bool, Float64, Int)
Real
```
This check is just to make sure the algorithm is working correctly.

Reply:
I don't understand, why we cannot use eltype in this test? What do we gain with typejoin(typeof.(...))?

Reply:
OK, i will change the tests.

Reply:
Made in [2f87e1d](https://github.com/JuliaML/TableTransforms.jl/pull/42/commits/2f87e1d78626a6982403f3de4ffd39eb107f4ab8)